### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by MrOmachi**
**In this milestone, I have done the following:**
- Implement classes and objects in Ruby.
- Implement encapsulation and inheritance with Ruby.
- Run a program using the command line.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './person'
+
+class Student < Person
+  def initialize(classroom)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```
Comment (DammyShittu) [inline student.rb]: - [ ] You are passing `age`, `name`, and `parent_permission` back to the Person class for creation but you are not passing them into the constructor. Kindly fix that:

  ```
    def initialize(age, classroom, name = "Unknown", parent_permission: true)
      super(age, name, parent_permission: parent_permission)
      @classroom = classroom
    end

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require_relative './person'
+class Teacher < Person
+  def initialize(specialization)
+    super(name, age, parent_permission)
+    @specialization = specialization
+  end
```
Comment (DammyShittu) [inline teacher.rb]: - [x] You are passing `age`, `name`, and `parent_permission` back to the Person class for creation but you are not passing them into the constructor. Kindly fix that.

- [ ] You also need to re-arrange your parameters in the `super` method to match the Person class constructor

  ```
  def initialize(age, specialization, name = "Unknown", parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization = specialization
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @MrOmachi,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @MrOmachi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**